File: client/src/app/models/review.ts
import { Avatar } from "./avatar"
import { Platforms } from "./platforms";
import { Poster } from "./poster";

export interface Review {
  id: number;
  reviewerId: number;
  reviewerUsername: string;
  reviewerAvatar: Avatar;
  gameId: number;
  gameTitle: string;
  gamePoster: Poster;
  content: string;
  reviewPosted: Date;
}

export interface ReviewMenu {
  id: number;
  posted: boolean;
  gameId: number;
  gameTitle: string;
  gamePlatforms: Platforms;
  gamePoster: Poster;
  content: string;
}

export interface ReviewForModeration {
  id: number;
  reviewerId: number;
  reviewerUsername: string;
  reviewerAvatar: Avatar;
  gameId: number;
  gameTitle: string;
  gamePoster: Poster;
  content: string;
  reviewPosted: Date;
  isApproved: boolean;
}

File: API/Entities/Game/Review.cs
namespace API.Entities;

public class Review
{
    public int Id { get; set; }
    public int ReviewerId { get; set; }
    public string ReviewerUsername { get; set; }
    public AppUser Reviewer { get; set; }
    public int GameId { get; set; }
    public string GameTitle { get; set; }
    public Game Game { get; set; }
    public string Content { get; set; }
    public DateTime ReviewPosted { get; set; } = DateTime.UtcNow;
    public bool IsApproved { get; set; }
}